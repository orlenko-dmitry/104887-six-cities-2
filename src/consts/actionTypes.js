export const SIGN_IN_SUCCESS = `SIGN_IN_SUCCESS`;
export const GET_USER_SUCCESS = `GET_USER_SUCCESS`;
export const FETCH_COMMENTS_PENDING = `FETCH_COMMENTS_PENDING`;
export const FETCH_COMMENTS_SUCCESS = `FETCH_COMMENTS_SUCCESS`;
export const FETCH_COMMENTS_ERROR = `FETCH_COMMENTS_ERROR`;
export const POST_COMMENTS_PENDING = `POST_COMMENTS_PENDING`;
export const POST_COMMENTS_SUCCESS = `POST_COMMENTS_SUCCESS`;
export const POST_COMMENTS_ERROR = `POST_COMMENTS_ERROR`;
export const FETCH_OFFERS_PENDING = `FETCH_OFFERS_PENDING`;
export const FETCH_OFFERS_SUCCESS = `FETCH_OFFERS_SUCCESS`;
export const FETCH_OFFERS_ERROR = `FETCH_OFFERS_ERROR`;
export const POST_FAVORITE_PENDING = `POST_FAVORITE_PENDING`;
export const POST_FAVORITE_SUCCESS = `POST_FAVORITE_SUCCESS`;
export const POST_FAVORITE_ERROR = `POST_FAVORITE_ERROR`;
export const SELECT_CITY = `SELECT_CITY`;
export const SORT_OFFERS_BY = `SORT_OFFERS_BY`;
export const OFFER_ON_HOVER = `OFFER_ON_HOVER`;
