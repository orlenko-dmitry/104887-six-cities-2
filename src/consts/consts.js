export const LeafletConst = {
  iconUrl: ({onHoverOfferId, selectedOfferId, currentId}) => {
    return selectedOfferId === currentId
    || onHoverOfferId === currentId ? `/img/pin-active.svg` : `/img/pin.svg`;
  },
  ICON_SIZE: [30, 30],
  TILE_LAYER: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  ATTRIBUTION: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`,
};

export const SortedBy = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const AppRoute = {
  ROOT: `/`,
  AUTH: `/login`,
  OFFER: `/offer`,
  FAVORITE: `/favorites`,
};

export const ResponseStatus = {
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};

export const MAX_REVIEWS = 10;

export const SortByList = [
  {
    id: 1,
    name: `Popular`,
  },
  {
    id: 2,
    name: `Price: low to high`,
  },
  {
    id: 3,
    name: `Price: high to low`,
  },
  {
    id: 4,
    name: `Top rated first`,
  },
];

export const AxiosConfig = {
  BASE_URL: ` https://htmlacademy-react-2.appspot.com/six-cities`,
  TIME_OUT: 5000,
  WITH_CREDENTIAL: true,
};

export const AsyncStatus = {
  PENDING: `PENDING`,
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
};

export const DateFormat = {
  DATE_TIME: `YYYY-MM-DD`,
  MOTH_YEAR: `MMMM YYYY`,
};

export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private Room`,
  HOUSE: `House`,
  HOTEL: `Hotel`,
};

export const AppCity = [
  {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.37454,
      "longitude": 4.897976,
      "zoom": 13
    }
  },
  {
    "name": `Brussels`,
    "location": {
      "latitude": 50.846557,
      "longitude": 4.351697,
      "zoom": 13
    }
  },
  {
    "name": `Cologne`,
    "location": {
      "latitude": 50.938361,
      "longitude": 6.959974,
      "zoom": 13
    }
  },
  {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13
    }
  },
  {
    "name": `Hamburg`,
    "location": {
      "latitude": 53.550341,
      "longitude": 10.000654,
      "zoom": 13
    }
  },
  {
    "name": `Dusseldorf`,
    "location": {
      "latitude": 51.225402,
      "longitude": 6.776314,
      "zoom": 13
    }
  }
];
