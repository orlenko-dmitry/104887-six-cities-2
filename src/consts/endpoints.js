export default ({
  login: `/login`,
  offers: `/hotels`,
  comments: (offerId) => `/comments/${offerId}`,
});
