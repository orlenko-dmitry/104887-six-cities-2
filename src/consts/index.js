export const LEAFLET_CONSTS = {
  iconUrl: ({onHoverOfferId, selectedOfferId, currentId}) => {
    return selectedOfferId === currentId
    || onHoverOfferId === currentId ? `img/pin-active.svg` : `img/pin.svg`;
  },
  ICON_SIZE: [30, 30],
  TILE_LAYER: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  ATTRIBUTION: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`,
};

export const SORTED_BY = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const SORT_BY_LIST = [
  {
    id: 1,
    name: `Popular`,
  },
  {
    id: 2,
    name: `Price: low to high`,
  },
  {
    id: 3,
    name: `Price: high to low`,
  },
  {
    id: 4,
    name: `Top rated first`,
  },
];

export const AXIOS_CONFIG = {
  BASE_URL: ` https://htmlacademy-react-2.appspot.com/six-cities`,
  TIME_OUT: 5000,
  WITH_CREDENTIAL: true,
};

export const ASYNC_STATUSES = {
  PENDING: `PENDING`,
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
};

export const DATE_FORMATS = {
  DATE_TIME: `YYYY-MM-DD`,
  MOTH_YEAR: `MMMM YYYY`,
};

export const OFFER_TYPE = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`,
};

export const APP_CITIES = [
  {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.37454,
      "longitude": 4.897976,
      "zoom": 13
    }
  },
  {
    "name": `Brussels`,
    "location": {
      "latitude": 50.846557,
      "longitude": 4.351697,
      "zoom": 13
    }
  },
  {
    "name": `Cologne`,
    "location": {
      "latitude": 50.938361,
      "longitude": 6.959974,
      "zoom": 13
    }
  },
  {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13
    }
  },
  {
    "name": `Hamburg`,
    "location": {
      "latitude": 53.550341,
      "longitude": 10.000654,
      "zoom": 13
    }
  },
  {
    "name": `Dusseldorf`,
    "location": {
      "latitude": 51.225402,
      "longitude": 6.776314,
      "zoom": 13
    }
  }
];
