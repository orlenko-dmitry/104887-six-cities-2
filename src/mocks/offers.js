const offers = [{
  "id": 1,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-01.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful & luxurious studio at great location`,
  "isFavorite": true,
  "isPremium": true,
  "rating": 4.8,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 4,
  "price": 120,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 2,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/room.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Wood and stone place`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 4.2,
  "type": `private room`,
  "bedrooms": 1,
  "max_adults": 2,
  "price": 80,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 3,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-02.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Nice, cozy, warm big bed apartmen`,
  "isFavorite": true,
  "isPremium": false,
  "rating": 4,
  "type": `apartment`,
  "bedrooms": 2,
  "max_adults": 3,
  "price": 70,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 4,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-03.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful luxurious apartment at great location`,
  "isFavorite": false,
  "isPremium": true,
  "rating": 4.9,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 6,
  "price": 200,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}];

export default offers;
