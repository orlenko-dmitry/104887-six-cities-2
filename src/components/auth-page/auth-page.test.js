import React from 'react';
import {shallow} from 'enzyme';

import AuthPage from './auth-page.jsx';
import offers from '../../mocks/offers.js';
import {AsyncStatus} from '../../consts/consts.js';

const city = offers[0].city;

it(`AuthPage renders correctly`, () => {
  const tree = shallow(
      <AuthPage
        city={city}
        userEmail=""
        userPassword=""
        user={null}
        userGetStatus={AsyncStatus.ERROR}
        onFormSubmit={() => {}}
        onEmailChange={() => {}}
        onPasswordChange={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
