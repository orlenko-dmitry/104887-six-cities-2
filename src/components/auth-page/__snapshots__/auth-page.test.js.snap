// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage renders correctly 1`] = `
<div
  className="page page--gray page--login"
>
  <PageHeader
    user={null}
  />
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          className="login__form form"
          onSubmit={[Function]}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="login__submit form__submit button"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <Link
            className="locations__item-link"
            to="/"
          >
            <span>
              Amsterdam
            </span>
          </Link>
        </div>
      </section>
    </div>
  </main>
</div>
`;
